### ISSUE TITLE
[BUG] TypeError on 'map' function when parsing monitors - Affects Version >= 10.10.0

### ISSUE THREAD

**Issue by samualblair** [f5devcentral/vscode-f5-flipper#36]
**Describe the bug**

TypeError when attempting to explore JSON (and therefore convert to anything) running Version >= 10.10.0
Unable to explore JSON (and therefore convert to anything) running Version 1.10.1 and 10.10.0 while working with 1.8

**To Reproduce**
1. Open workspace
2. Right on ns.conf file, use F5 Flipper "Explore ADC/NS (.conf/tgz)
3. Initial parsing appears to work, applications are shown.
4. Select one of the applications, click on the "{ }" , nothing is shown, output logs indicate TypeError occurred

**Expected behavior**
Expect a json to be generated and displayed in a new editor tab/window.

**Screenshots**
See logs below.

```
1 [2024-05-23T17:52:04.925Z] [INFO]: refreshing ns diagnostic rules and tree view
1 [2024-05-23T17:52:04.925Z] [INFO]: loading ns diagnosics rules file
1 [2024-05-23T17:52:13.064Z] [ERROR]: --- unhandledRejection --- [TypeError: Cannot read properties of undefined (reading 'map')
	at mungeNS2FAST (/Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/ns2FastParams.js:109:107)
	at NsCfgProvider.<anonymous> (/Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/nsCfgViewProvider.js:438:73)
	at Generator.next (<anonymous>)
	at /Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/nsCfgViewProvider.js:15:71
	at new Promise (<anonymous>)
	at __awaiter (/Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/nsCfgViewProvider.js:11:12)
	at NsCfgProvider.render (/Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/nsCfgViewProvider.js:427:16)
	at /Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/extension.js:191:52
	at Generator.next (<anonymous>)
	at /Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/extension.js:31:71
	at new Promise (<anonymous>)
	at __awaiter (/Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/extension.js:27:12)
	at /Users/michaelj/.vscode/extensions/f5devcentral.vscode-f5-flipper-1.10.1/out/extension.js:187:100
	at r.h (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:153:189465)
	at r.$executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:153:190325)
	at c.S (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:150:5505)
	at c.Q (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:150:5271)
	at c.M (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:150:4361)
	at c.L (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:150:3579)
	at a.value (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:150:2227)
	at o.y (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:82:660)
	at o.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:82:877)
	at u.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:107:14175)
	at a.value (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:176:8023)
	at o.y (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:82:660)
	at o.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:82:877)
	at u.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:107:14175)
	at MessagePortMain.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:176:6303)
	at MessagePortMain.emit (node:events:517:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:2285)]
```

**Desktop (please complete the following information):**
 - OS: macOS 14.5

 - VS Code: Version: 1.89.1
Commit: dc96b837cf6bb4af9cd736aa3af08cf8279f7685
Date: 2024-05-07T05:14:32.757Z (2 wks ago)
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin arm64 23.5.0

 - F5 Flipper 10.10.1 and 10.10.0 have symptoms.
 - F5 Flipper 10.8.0 Does not have the symptom.

**Comment by samualblair** [comment]
I'm thinking there must be some parameter in my NS config file that is not mapping well with the new code versions, because the "Lead Example / Test NS Config" seems to work as expected.

**Comment by samualblair** [comment]
Offering a pull request for the workaround, [https://github.com/f5devcentral/vscode-f5-flipper/pull/37](url)

**Comment by samualblair** [comment]
Canceling that, found the issue after a few more minutes of looking.

The object nsFastJson was missing the monitors.

So this:
```
        const nsFastJson = {
            tenant_name: nsApp.name,
            app_name: nsApp.name,
            type: nsApp.type,
            protocol: nsApp.protocol,
            virtual_address: nsApp.ipAddress,
            virtual_port: nsApp.port === '*' ? '0' : nsApp.port,
            pool_members: []
        };
```

Should be:
```
        const nsFastJson = {
            tenant_name: nsApp.name,
            app_name: nsApp.name,
            type: nsApp.type,
            protocol: nsApp.protocol,
            virtual_address: nsApp.ipAddress,
            virtual_port: nsApp.port === '*' ? '0' : nsApp.port,
            pool_members: [],
            monitors: []
        };
```

This seems to fix the issue and not prevent the monitors from coming over.

**Comment by samualblair** [comment]
Offering a pull request with fix , [38](https://github.com/f5devcentral/vscode-f5-flipper/pull/38)

**Comment by DumpySquare** [comment]
this fix will in the pending v1.11.0 release
